src/main/java/com/slack/kaldb/chunk/ChunkManager.java

```diff
@@ -128,12 +129,20 @@ public ChunkManager(
     this.rolloverFutureTimeoutMs = rollOverFutureTimeoutMs;
     stopIngestion = false;
     activeChunk = null;
+
     LOG.info(
         "Created a chunk manager with prefix {} and dataDirectory {}",
         chunkDataPrefix,
         dataDirectory);
   }
 
+  /*
+     One day we will have to think about rate limiting/backpressure and we will revisit this so it could potentially reject threads if the pool is full
+  */
+  private static ExecutorService queryThreadPool() {
+    return Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
```

Comment:
This probably would be better to be a configuration value instead of just using the runtime. The `availableProcessors` historically hasn't been super great at correctly detecting vCPUs. Additionally when deploying to different hardware configurations it can make detecting performance issues more difficult.

Reply:
that makes sense! We could use `Runtime.getRuntime().availableProcessors()` as default if the the value isn't configured

But I really want to stop using a thread pool here altogether - i.e go away from the chunk manager aggregating locally to making 1 gRPC call per chunk - it makes it easier to handle errors etc

So I will do that shortly after these PRs